tests/runtime/regression

```diff
@@ -0,0 +1,11 @@
+# https://github.com/iovisor/bpftrace/pull/566#issuecomment-487295113
+NAME codegen_struct_save_nested
+RUN bpftrace -e 'struct Foo { int m; struct { int x; int y; } bar; int n; } BEGIN { @foo = (Foo)0; @bar = @foo.bar; @x = @foo.bar.x; printf("done\n"); exit(); }'
+EXPECT done
+TIMEOUT 1
+
+# https://github.com/iovisor/bpftrace/issues/759
+NAME ntop_map_printf
+RUN bpftrace -e 'BEGIN { @a = ntop(2, 0); printf("%s", @a); @b = ntop(10, 0); exit() } END { printf("%s", @b); clear(@a); clear(@b) }'
+EXPECT 0.0.0.0::900:0:200:1800
```

Comment:
Where's the `900:0:200:1800` coming from?

Reply:
This was the value I was always getting when running this script, but now that I think of it this is not very reliable. I'll add a testprog with IPv4 and IPv6 addresses properly defined.

Reply:
Good, apparently this uncovered some more bugs :)

Reply:
Ok, I rewrote the test to use a `testprog` binary, which will give us predictable results. Unfortunately I found out that ntop for IPv6 doesn't work well with arguments from uprobe/USDT, because we need an array of 8-bit integers, and uprobe/USDT arguments are always treated as 64-bit integers. If ntop accepted pointers (#779) or strings (#780 ) it would work fine. I opened #781 to make ntop accept string arguments, but I don't think we should block this PR, since it fixes a bug with ntop, so I commented the test and I'll uncomment it once #781 lands.

Reply:
But shouldn't `ntop(0)` just return the "zero" IP address? I would have thought the test you had before should work ok, but just print `0.0.0.0::` (zero IPv4 address, then zero IPv6 address)

Reply:
Ah, yes, you're right. Apparently this is happening due to https://github.com/iovisor/bpftrace/issues/718 :(

```
# bpftrace -e 'BEGIN { @ = ntop(10, 0); printf("%s\n", @); }'
Attaching 1 probe...
::
^C

@: ::b0c8:c3ee:d56:0:2000:0

# bpftrace -e 'BEGIN { @ = ntop(10, 0); printf("%s\n", @); exit()}'
Attaching 1 probe...
::900:0:200:1800


@: ::c082:2dc3:5e55:0:2000:0

# bpftrace -e 'BEGIN { @ = ntop(10, 0); printf("%s\n", @); clear(@)}'
Attaching 1 probe...
::900:0:200:1800
^C
```

Why it's happening for integers but not for arrays? I have no idea, but I'll try to find out.

Reply:
Found the problem: we shouldn't allow integer arguments for IPv6 values. 8-byte integers leave half of the 16-byte IPv6 address undefined so it's just whatever was in memory before.

Reply:
That can be blocked in another pull request (like #781) though

Reply:
> we shouldn't allow integer arguments for IPv6 values

IDK, I think we can make it work for integer arguments as well. Not sure if it's worth the trouble though, since IPv6 will usually be an array.

(FWIW I was about to open an issue for us to either make it work or to block it's use in the semantic analyser)